From: Marcelo Cerri <marcelo.cerri@canonical.com>
Subject: Fix build for 4.20

BugLink: https://bugs.launchpad.net/bugs/1813068

Signed-off-by: Marcelo Cerri <marcelo.cerri@canonical.com>
Signed-off-by: Connor Kuehl <connor.kuehl@canonical.com>
---
--- a/os_dep/linux/ioctl_cfg80211.c
+++ b/os_dep/linux/ioctl_cfg80211.c
@@ -358,7 +358,11 @@
 
 static u64 rtw_get_systime_us(void)
 {
-#if (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,39))
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(4,20,0))
+	struct timespec64 ts;
+	ktime_get_boottime_ts64(&ts);
+	return ((u64)ts.tv_sec*1000000) + ts.tv_nsec / 1000;
+#elif (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,39))
 	struct timespec ts;
 	get_monotonic_boottime(&ts);
 	return ((u64)ts.tv_sec*1000000) + ts.tv_nsec / 1000;
